### PR TITLE
[release/8.0] Chain runtime .Msi's instead of .Exe's in hosting bundle

### PR DESCRIPTION
**Pull request by wtgodbe**
Attempt to fix https://github.com/dotnet/aspnetcore/issues/55950

Test build: https://dev.azure.com/dnceng/internal/_build/results?buildId=2482344&view=results
Test build w/ patch incremented: https://dev.azure.com/dnceng/internal/_build/results?buildId=2482415&view=results
Test build w/ patch decremented: https://dev.azure.com/dnceng/internal/_build/results?buildId=2483057&view=results

### CODE REVIEW

File: src/Installers/Windows/WindowsHostingBundle/Product.targets [hunk 1]
```diff
@@ -7,22 +7,30 @@
...
   <ItemGroup>
-    <Platforms Include="x64;x86;arm64" />
-    <RuntimeInstallers Include="$(DepsPath)dotnet-runtime-$(MicrosoftNETCoreAppRuntimeVersion)-win-x64.exe">
+    <RuntimeInstallers Include="$(DepsPath)dotnet-runtime-$(MicrosoftNETCoreAppRuntimeVersion)-win-x64.msi">
       <TargetPlatform>x64</TargetPlatform>
       <BundleNameProperty>DotNetRedistLtsInstallerx64</BundleNameProperty>
       <Version>$(MicrosoftNETCoreAppRuntimeVersion)</Version>
     </RuntimeInstallers>
-    <RuntimeInstallers Include="$(DepsPath)dotnet-runtime-$(MicrosoftNETCoreAppRuntimeVersion)-win-x86.exe">
+    <RuntimeInstallers Include="$(DepsPath)dotnet-runtime-$(MicrosoftNETCoreAppRuntimeVersion)-win-x86.msi">
```
Comment (joeloff) [inline src/Installers/Windows/WindowsHostingBundle/Product.targets:15]: We need two additional MSIs for runtime, similar to what we do in desktop. Both the host and hostfxr MSIs need to be included

File: src/Installers/Windows/WindowsHostingBundle/SharedFramework.wxs [hunk 2]
```diff
@@ -24,40 +24,29 @@
...
                         Name="$(var.SharedFxRedistInstallerarm64)"
                         Compressed="yes"
                         Vital="yes"
                         InstallCondition="(NativeMachine=&quot;$(var.NativeMachine_arm64)&quot;) AND (NOT OPT_NO_SHAREDFX OR OPT_NO_SHAREDFX=&quot;0&quot;)"
-                        InstallCommand="/quiet /norestart"
-                        RepairCommand="/quiet /norestart /repair"
-                        Permanent="yes"
-                        DetectCondition="SharedFxRedistProductVersion_arm64 = v$(var.SharedFxInstallerProductVersionarm64)">
-            </ExePackage>
+                        Permanent="yes">
```
Comment (joeloff) [inline src/Installers/Windows/WindowsHostingBundle/SharedFramework.wxs:32]: Think we need to remove the Permanent attribute too. When set, the MSI is marked as a system component so it won't remove it.

File: src/Installers/Windows/WindowsHostingBundle/Product.targets (hunk 1 above)
Comment (wtgodbe) [reply src/Installers/Windows/WindowsHostingBundle/Product.targets:13]: Added

Comment (wtgodbe) [reply src/Installers/Windows/WindowsHostingBundle/Product.targets:13]: Also realized we don't need the ProductSearch anymore, removed those too

File: src/Installers/Windows/WindowsHostingBundle/SharedFramework.wxs (hunk 2 above)
Comment (wtgodbe) [reply src/Installers/Windows/WindowsHostingBundle/SharedFramework.wxs]: Done

**Comment by wtgodbe** [comment]
New build https://dev.azure.com/dnceng/internal/_build/results?buildId=2488177&view=results

**Review by joeloff — APPROVED** [review]
Do you have an internal build of this so I can look at a copy of the install?

**Comment by wtgodbe** [comment]
> Do you have an internal build of this so I can look at a copy of the install?

https://dev.azure.com/dnceng/internal/_build/results?buildId=2488228&view=results is almost finished

**Pull request merged**